game-core/src/main/java/games/strategy/engine/chat/JavaSocketChatTransmitter.java

```diff
@@ -0,0 +1,104 @@
+package games.strategy.engine.chat;
+
+import games.strategy.engine.lobby.PlayerName;
+import games.strategy.engine.message.MessageContext;
+import games.strategy.engine.message.RemoteName;
+import games.strategy.net.Messengers;
+import java.util.Map;
+
+/** Chat transmitter that sends and receives messages over Java NIO sockets. */
+class JavaSocketChatTransmitter implements ChatTransmitter {
+  private final PlayerName playerName;
+  private final Messengers messengers;
+
+  private final IChatChannel chatChannelSubscriber;
+
+  private final String chatName;
+  private final String chatChannelName;
+
+  JavaSocketChatTransmitter(
+      final ChatClient chatClient, final String chatName, final Messengers messengers) {
+    this.playerName = messengers.getLocalNode().getPlayerName();
+    this.messengers = messengers;
+    this.chatName = chatName;
+    this.chatChannelName = ChatController.getChatChannelName(chatName);
+    chatChannelSubscriber = chatChannelSubscriber(chatClient);
+  }
+
+  private IChatChannel chatChannelSubscriber(final ChatClient chatClient) {
+    return new IChatChannel() {
+      @Override
+      public void chatOccurred(final String message) {
+        chatClient.messageReceived(message);
+      }
+
+      @Override
+      public void slapOccurred(final PlayerName slappedPlayer) {
+        final PlayerName slapper = MessageContext.getSender().getPlayerName();
+        if (slappedPlayer.equals(playerName)) {
+          chatClient.slappedBy(slapper);
+        } else {
+          chatClient.eventMessage(slappedPlayer + " was slapped by " + slapper);
+        }
+      }
+
+      @Override
+      public void speakerAdded(final ChatParticipant chatParticipant) {
+        chatClient.participantAdded(chatParticipant);
+      }
+
+      @Override
+      public void speakerRemoved(final PlayerName playerName) {
+        chatClient.participantRemoved(playerName);
+      }
+
+      @Override
+      public void ping() {}
```

Comment:
I was focused on getting the class split up and not dependent on Messengers, it was a pretty difficult update and did not go smoothly the first 2 times I tried it. It would be nice to see the method removed for the simplification, but without contrary evidence we should assume it's needed.

Reply:
Did some research on this, `ping()` was introduced in: 0c9f4f675679cc2618bc6a93d1bd2f7dcecd69c0, with comment:

> Creating a simple Ping method inside of the Chat Controller, that will ping all computers once every minute, hopefully improving everyone's connections. (Sean Bridges & veqryn)


We do set the TCP keep-alive option:
>     socket.setKeepAlive(true);

This SO seems to mention some of the gotchyas: https://stackoverflow.com/questions/1480236/does-a-tcp-socket-connection-have-a-keep-alive

> Gotchas
2 Hour Default

> Since the process doesn't start until a connection has been idle for two hours by default, stale TCP connections can linger for a very long time before being pruned. This can be especially harmful for expensive connections such as database connections.
Keep-Alive is Optional

> According to RFC 1122 4.2.3.6, responding to and/or relaying TCP Keep-Alive packets is optional

--------------

If lobby chat is converted to http, the keep-alive mechanism there will certainly send packets frequently enough to avoid any of the above limits. In-game chats should likely get traffic, at least if a game is started. I would be concerned a about 'staging' screens where players queue up and do a long joining period, eg: someone hosts and then over the course of hours players slowly join up without any conversation happening. 

On the other hand, TCP keep-alive is supposed to keep the connection open without us manually sending messages. I suspect we could probably drop this feature, but I would be afraid we'd find out the hard way where/how it was needed.

@RoiEXLab any thoughts based on that research?